src/simulator/calculations/Force.cpp

```diff
@@ -0,0 +1,110 @@
+#include "Force.h"
+#include "../particle/ParticleContainer.h"
+#include "utils/ArrayUtils.h"
+
+void Force::run(LinkedCellContainer &particles, ForceType type,
+                OPTIONS OPTION) {
+  switch (type) {
+  case LENNARD_JONES:
+    lennard_jones(particles, OPTION);
+    break;
+  case VERLET:
+    verlet(particles, OPTION);
+    break;
+  }
+}
+
+void Force::lennard_jones(LinkedCellContainer &particles, OPTIONS OPTION) {
+  if (OPTION == OPTIONS::NONE) {
+
+    for (auto &p : particles) {
+      p.updateOldF(p.getF());
+      p.updateF(0, 0, 0);
+    }
+
+    for (auto it = particles.pair_begin(); it != particles.pair_end(); ++it) {
+      auto r12 = it->second->getX() - it->first->getX();
+      double distance = ArrayUtils::L2Norm(r12);
+
+      double totalForce;
+      double term = SIGMA / distance;
+      double term6 = pow(term, 6);
+      double term12 = pow(term, 12);
+      totalForce = 24 * EPSILON * (term6 - 2 * term12) / distance;
+
+      auto force = (totalForce / distance) * r12;
+
+      it->first->updateF(it->first->getF() + force);
+      it->second->updateF(it->second->getF() - force);
+    }
+  } else {
+    for (auto &p : particles) {
+      p.updateOldF(p.getF());
+      p.updateF(0, 0, 0);
+    }
+
+    for (auto &particle : particles) {
+      for (auto neighbour : particles.get_neighbours(particle)) {
```

Comment:
it seems like you have missed considering the interaction from particles in the same cells as the current particle. the `get_neighbours()` only returns particle in the neighbouring cells and not the same cell.